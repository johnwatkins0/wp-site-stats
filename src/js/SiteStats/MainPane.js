/* eslint camelcase: 0 */
import React from 'react';
import format from 'date-fns/format';
import get from 'lodash.get';

const getCommentExcerpt = ( comment ) => {
	const words = comment.split( ' ' );
	if ( 15 > words.length ) {
		return words.join( ' ' );
	}

	return `${words.slice( 0, 15 ).join( ' ' )} ...`;
};

const Stat = ({ title, number }) => (
	<div className="MainPane__stat-container">
		<div className="MainPane__stat">
			<h4 className="MainPane__stat-title">
				{title}
			</h4>
			<div className="MainPane__stat-number">
				{number}
			</div>
		</div>
	</div>
);

const MainPane = ({
	site_title = '&nbsp;',
	site_url,
	user_count = '-',
	post_count,
	page_count,
	latest_post,
	latest_comment,
	comment_count,
	activeSiteCreatedDate,
	secondsToRefresh,
	shouldRefresh
}) => (
	<article className="MainPane">
		<header className="MainPane__header">
			<h2 dangerouslySetInnerHTML={ { __html: site_title } } />
			<div className="MainPane__site-info">
				<div>
					<span className="MainPane__site-info-key">
						URL:
					</span>
					{' '}
					<a href={site_url}>
						{site_url}
					</a>
				</div>
				<div>
					<span className="MainPane__site-info-key">
						Created:
					</span>
					{' '}
					{format( activeSiteCreatedDate, 'MMMM D, YYYY' )}
				</div>
			</div>
		</header>
		<section className="MainPane__section">
			<h3>By the Numbers</h3>
			<div  className="MainPane__stats">
				<Stat title="Posts" number={get( post_count, 'publish', '-' )} />
				<Stat title="Pages" number={get( page_count, 'publish', '-' )} />
				<Stat title="Users" number={user_count} />
				<Stat title="Comments" number={get( comment_count, 'approved', '-' )} />
			</div>
		</section>
		<section className="MainPane__section MainPane__latest-post">
			<h3>
				Latest Post
			</h3>
			<div className="MainPane__latest-post-link">
				<a
					href={get( latest_post, 'link', '' )}
					dangerouslySetInnerHTML={
						{
							__html: get(
								latest_post, [ 'title', 'rendered' ], '&nbsp'
							) || '(no-title)'
						}
					}
				/>
			</div>
		</section>
		<section className="MainPane__section MainPane__latest-comment">
			<h3>
				Latest Comment
			</h3>
			<div className="MainPane__latest-comment-link">
				<a href={ get( latest_comment, 'link' )}>
					{
						get( latest_comment, 'author_name', false ) &&
						`${latest_comment.author_name} `
					}
					{
						get( latest_comment, 'date', false ) &&
						`on ${format( latest_comment.date, 'MMMM D, YYYY' )}
						at ${format( latest_comment.date, 'h:mm aa' )}`
					}
				</a>
			</div>
		</section>
		{shouldRefresh && <footer className="MainPane__footer">
			Refreshing in {secondsToRefresh} second{1 === secondsToRefresh ? '' : 's'}.
		</footer>}
	</article>
);

export default MainPane;
