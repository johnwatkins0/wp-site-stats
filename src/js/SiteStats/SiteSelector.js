import React from 'react';

const SiteSelector = ({ activeSiteId, sites, setActiveSite }) => (
	<label>
		Sites:{' '}
		<select
			className="SiteSelector"
			value={activeSiteId}
			onChange={event => {
				setActiveSite( event.target.value );
			}}
		>

			{sites.map( ({ id, domain, path }) => (
				<option
					key={id}
					value={id}
				>
					{`${domain}${path}`}
				</option>
			) )}
		</select>
	</label>
);

export default SiteSelector;
