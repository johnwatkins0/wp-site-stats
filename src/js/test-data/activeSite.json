{
    "site_title": "My Site",
    "site_url": "http://wpmu/my-site",
    "user_count": 1,
    "post_count": {
        "publish": "1",
        "future": 0,
        "draft": 0,
        "pending": 0,
        "private": 0,
        "trash": 0,
        "auto-draft": 0,
        "inherit": 0
    },
    "page_count": {
        "publish": "1",
        "future": 0,
        "draft": 0,
        "pending": 0,
        "private": 0,
        "trash": 0,
        "auto-draft": 0,
        "inherit": 0
    },
    "comment_count": {
        "approved": "1",
        "spam": 0,
        "trash": 0,
        "post-trashed": 0,
        "total_comments": 1,
        "all": 1,
        "moderated": 0
    },
    "latest_post": {
        "id": 1,
        "date": "2018-04-02T14:29:11",
        "date_gmt": "2018-04-02T14:29:11",
        "guid": {
            "rendered": "http://wpmu/my-site/?p=1"
        },
        "modified": "2018-04-02T14:29:11",
        "modified_gmt": "2018-04-02T14:29:11",
        "slug": "hello-world",
        "status": "publish",
        "type": "post",
        "link": "http://wpmu/my-site/2018/04/02/hello-world/",
        "title": {
            "rendered": "Hello world!"
        },
        "content": {
            "rendered": "<p>Welcome to <a href=\"http://wpmu/\">Colby College Sites</a>. This is your first post. Edit or delete it, then start blogging!</p>\n",
            "protected": false
        },
        "excerpt": {
            "rendered": "<p>Welcome to Colby College Sites. This is your first post. Edit or delete it, then start blogging!</p>\n",
            "protected": false
        },
        "author": 1,
        "featured_media": 0,
        "comment_status": "open",
        "ping_status": "open",
        "sticky": false,
        "template": "",
        "format": "standard",
        "meta": [],
        "categories": [
            1
        ],
        "tags": [],
        "revisions": null
    },
    "latest_comment": {
        "id": 1,
        "post": 1,
        "parent": 0,
        "author": 0,
        "author_name": "A WordPress Commenter",
        "author_url": "http://wpmu/",
        "date": "2018-04-02T14:29:11",
        "date_gmt": "2018-04-02T14:29:11",
        "content": {
            "rendered": "<p>Hi, this is a comment.<br />\nTo get started with moderating, editing, and deleting comments, please visit the Comments screen in the dashboard.<br />\nCommenter avatars come from <a href=\"https://gravatar.com\">Gravatar</a>.</p>\n"
        },
        "link": "http://wpmu/my-site/2018/04/02/hello-world/#comment-1",
        "status": "approved",
        "type": "comment",
        "author_avatar_urls": {
            "24": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=24&d=mm&r=g",
            "48": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=48&d=mm&r=g",
            "96": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=96&d=mm&r=g"
        },
        "meta": []
    }
}