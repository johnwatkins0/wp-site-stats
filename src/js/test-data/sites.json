[
    {
        "id": "1",
        "domain": "wpmu",
        "path": "/",
        "registered": "2018-04-02 14:28:19"
    },
    {
        "id": "2",
        "domain": "wpmu",
        "path": "/my-site/",
        "registered": "2018-04-02 14:29:11"
    },
    {
        "id": "3",
        "domain": "wpmu",
        "path": "/another-site/",
        "registered": "2018-04-02 14:29:27"
    },
    {
        "id": "4",
        "domain": "wpmu",
        "path": "/cool-site/",
        "registered": "2018-04-03 17:13:22"
    },
    {
        "id": "5",
        "domain": "wpmu",
        "path": "/another-cool-site/",
        "registered": "2018-04-03 17:13:32"
    },
    {
        "id": "6",
        "domain": "wpmu",
        "path": "/better-site/",
        "registered": "2018-04-03 17:13:44"
    },
    {
        "id": "7",
        "domain": "wpmu",
        "path": "/best-site/",
        "registered": "2018-04-03 17:13:53"
    }
]