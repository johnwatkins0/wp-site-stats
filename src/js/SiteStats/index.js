import React from 'react';

import SiteStats from './SiteStats';

import { setActiveSiteId } from '../store/actions';
import {
	getSites,
	getSiteData,
	getActiveSiteId,
	getSiteCreatedDate,
	getSecondsToRefresh,
	getShouldRefresh
} from '../store/selectors';

const mapStateToProps = ( state ) => {
	const activeSiteId = getActiveSiteId( state );
	return {
		sites: getSites( state ),
		activeSiteId,
		activeSite: getSiteData( state, activeSiteId ),
		activeSiteCreatedDate: getSiteCreatedDate( state, activeSiteId ),
		secondsToRefresh: getSecondsToRefresh( state ),
		shouldRefresh: getShouldRefresh( state )
	};
};

const mapDispatchToProps = ( dispatch ) => ({
	setActiveSite: ( siteId ) => dispatch( setActiveSiteId( siteId ) )
});

const ConnectedSiteStats = ({ state, dispatch }) => (
	<SiteStats
		{...mapStateToProps( state )}
		{...mapDispatchToProps( dispatch )}
	/>
);

export default ConnectedSiteStats;
