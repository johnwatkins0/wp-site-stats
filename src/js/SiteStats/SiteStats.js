import React from 'react';

import MainPane from './MainPane';
import SiteSelector from './SiteSelector';

const SiteStats = ({
	sites,
	activeSiteId,
	activeSite,
	setActiveSite,
	activeSiteCreatedDate,
	secondsToRefresh,
	shouldRefresh
}) => (
	<div className="SiteStats">
		<div className="SiteStats__header">
			<h1>Site Stats</h1>
			{0 < sites.length && <SiteSelector
				sites={sites}
				activeSiteId={activeSiteId}
				setActiveSite={setActiveSite}
			/>}
		</div>
		<div className="SiteStats__body">
			<MainPane
				{...activeSite}
				activeSiteCreatedDate={activeSiteCreatedDate}
				shouldRefresh={shouldRefresh}
				secondsToRefresh={secondsToRefresh}
			/>
		</div>
	</div>
);

export default SiteStats;
