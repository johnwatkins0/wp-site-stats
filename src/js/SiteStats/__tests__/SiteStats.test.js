import React from 'react';
import renderer from 'react-test-renderer';

import SiteStats from '../SiteStats';

import sites from '../../test-data/sites.json';
import activeSite from '../../test-data/activeSite.json';

test( 'SiteStats component', () => {
	const tree = renderer.create(
		<SiteStats
			loading={false}
			sites={sites}
			activeSite={activeSite}
			activeSiteId={activeSite.id}
			setActiveSiteId={( id ) => null}
		/>
	).toJSON();
	expect( tree ).toMatchSnapshot();
});
