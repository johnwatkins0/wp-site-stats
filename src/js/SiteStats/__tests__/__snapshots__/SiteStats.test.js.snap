// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteStats component 1`] = `
<div
  className="SiteStats"
>
  <div
    className="SiteStats__header"
  >
    <h1>
      Site Stats
    </h1>
    <label>
      Sites:
       
      <select
        className="SiteSelector"
        onChange={[Function]}
        value={undefined}
      >
        <option
          value="1"
        >
          wpmu/
        </option>
        <option
          value="2"
        >
          wpmu/my-site/
        </option>
        <option
          value="3"
        >
          wpmu/another-site/
        </option>
        <option
          value="4"
        >
          wpmu/cool-site/
        </option>
        <option
          value="5"
        >
          wpmu/another-cool-site/
        </option>
        <option
          value="6"
        >
          wpmu/better-site/
        </option>
        <option
          value="7"
        >
          wpmu/best-site/
        </option>
      </select>
    </label>
  </div>
  <div
    className="SiteStats__body"
  >
    <article
      className="MainPane"
    >
      <header
        className="MainPane__header"
      >
        <h2
          dangerouslySetInnerHTML={
            Object {
              "__html": "My Site",
            }
          }
        />
        <div
          className="MainPane__site-info"
        >
          <div>
            <span
              className="MainPane__site-info-key"
            >
              URL:
            </span>
             
            <a
              href="http://wpmu/my-site"
            >
              http://wpmu/my-site
            </a>
          </div>
          <div>
            <span
              className="MainPane__site-info-key"
            >
              Created:
            </span>
             
            Invalid Date
          </div>
        </div>
      </header>
      <section
        className="MainPane__section"
      >
        <h3>
          By the Numbers
        </h3>
        <div
          className="MainPane__stats"
        >
          <div
            className="MainPane__stat-container"
          >
            <div
              className="MainPane__stat"
            >
              <h4
                className="MainPane__stat-title"
              >
                Posts
              </h4>
              <div
                className="MainPane__stat-number"
              >
                1
              </div>
            </div>
          </div>
          <div
            className="MainPane__stat-container"
          >
            <div
              className="MainPane__stat"
            >
              <h4
                className="MainPane__stat-title"
              >
                Pages
              </h4>
              <div
                className="MainPane__stat-number"
              >
                1
              </div>
            </div>
          </div>
          <div
            className="MainPane__stat-container"
          >
            <div
              className="MainPane__stat"
            >
              <h4
                className="MainPane__stat-title"
              >
                Users
              </h4>
              <div
                className="MainPane__stat-number"
              >
                1
              </div>
            </div>
          </div>
          <div
            className="MainPane__stat-container"
          >
            <div
              className="MainPane__stat"
            >
              <h4
                className="MainPane__stat-title"
              >
                Comments
              </h4>
              <div
                className="MainPane__stat-number"
              >
                1
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        className="MainPane__section MainPane__latest-post"
      >
        <h3>
          Latest Post
        </h3>
        <div
          className="MainPane__latest-post-link"
        >
          <a
            dangerouslySetInnerHTML={
              Object {
                "__html": "Hello world!",
              }
            }
            href="http://wpmu/my-site/2018/04/02/hello-world/"
          />
        </div>
      </section>
      <section
        className="MainPane__section MainPane__latest-comment"
      >
        <h3>
          Latest Comment
        </h3>
        <div
          className="MainPane__latest-comment-link"
        >
          <a
            href="http://wpmu/my-site/2018/04/02/hello-world/#comment-1"
          >
            A WordPress Commenter 
            on April 2, 2018
						at 2:29 p.m.
          </a>
        </div>
      </section>
    </article>
  </div>
</div>
`;
